project1-sps/script.js

```diff
@@ -1,4 +1,177 @@
+var userWinCount = 0;
+var pcWinCount = 0;
+var drawCount = 0;
+var usernameInputCount = 0;
+var username = 0;
+var message = 0;
+
+// counting rounds and win lose percentage
+var roundsPlayed = 0;
+var winLoseRoundsPlayed = 0;
+
+var chooseSPS = function () {
+  var randomNumber = Math.random() * 3;
+  var foodNumber = Math.floor(randomNumber);
+  return foodNumber + 1;
+};
+
+var usernameInputCheck = function (userSPS) {
+  //1st time input for username, so username input count is still 0
+  if (usernameInputCount < 1) {
+    username = userSPS;
+    var messageUsername = `Welcome ${username}. <br> Let's start playing scissors, paper, stones. `;
+  }
+  usernameInputCount += 1;
+  console.log(`usernameInputCheck message1 ${messageUsername}`);
+  return messageUsername;
```

Comment:
Interesting way of controlling the game mode! However, in the future as number of game modes increase, it might be more productive to store them in a string and label what the game modes actually represent instead of just 0, 1, 2, 3.